presto-main-base/src/test/java/com/facebook/presto/sql/query/TestColumnMask.java

```diff
@@ -0,0 +1,382 @@
+                    new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                    "custkey",
+                    USER,
+                    new ViewExpression(USER, Optional.empty(), Optional.empty(), "-custkey"));
+            assertions.assertQuery("SELECT custkey FROM orders WHERE orderkey = 1", "VALUES BIGINT '-370'");
+        });
+
+        assertions.executeExclusively(() -> {
+            accessControl.reset();
+            accessControl.columnMask(
+                    new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                    "custkey",
+                    USER,
+                    new ViewExpression(USER, Optional.empty(), Optional.empty(), "NULL"));
+            assertions.assertQuery("SELECT custkey FROM orders WHERE orderkey = 1", "VALUES CAST(NULL AS BIGINT)");
+        });
+    }
+
+    @Test
+    public void testMultipleMasksOnDifferentColumns()
+    {
+        assertions.executeExclusively(() -> {
+            accessControl.reset();
+            accessControl.columnMask(
+                    new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                    "custkey",
+                    USER,
+                    new ViewExpression(USER, Optional.empty(), Optional.empty(), "-custkey"));
+
+            accessControl.columnMask(
+                    new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                    "orderstatus",
+                    USER,
+                    new ViewExpression(USER, Optional.empty(), Optional.empty(), "'X'"));
+
+            assertions.assertQuery("SELECT custkey, orderstatus FROM orders WHERE orderkey = 1", "VALUES (BIGINT '-370', 'X')");
+        });
+    }
+
+    @Test
+    public void testCoercibleType()
+    {
+        assertions.executeExclusively(() -> {
+            accessControl.reset();
+            accessControl.columnMask(
+                    new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                    "clerk",
+                    USER,
+                    new ViewExpression(USER, Optional.empty(), Optional.empty(), "CAST(clerk AS VARCHAR(5))"));
+            assertions.assertQuery("SELECT clerk FROM orders WHERE orderkey = 1", "VALUES CAST('Clerk' AS VARCHAR(15))");
+        });
+    }
+
+    @Test
+    public void testSubquery()
+    {
+        // uncorrelated
+        assertions.executeExclusively(() -> {
+            accessControl.reset();
+            accessControl.columnMask(
+                    new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                    "clerk",
+                    USER,
+                    new ViewExpression(USER, Optional.of(CATALOG), Optional.of("tiny"), "(SELECT cast(max(name) AS VARCHAR(15)) FROM nation)"));
+            assertions.assertQuery("SELECT clerk FROM orders WHERE orderkey = 1", "VALUES CAST('VIETNAM' AS VARCHAR(15))");
+        });
+
+        // correlated
+        assertions.executeExclusively(() -> {
+            accessControl.reset();
```

Comment:
These tests use `accesscontrol.reset()` while others do not? Is it required?

Reply:
It should be called in each test that adds filters or masks to access control - which ones did you see that don't have it?

Reply:
Hmm I swear I saw some without it. I don't see the same issue any more. Forget about this comment I guess